tensorrt_llm/_torch/auto_deploy/transform/library/visualization.py

```diff
@@ -87,24 +30,57 @@ def add_outputs_metadata(self, fx_node: torch.fx.node.Node, node: GraphNode):
 
 @TransformRegistry.register("visualize_namespace")
 class VisualizeNamespace(BaseTransform):
+    """Transform to visualize the graph using Model Explorer.
+
+    This transform exports the graph module to an ExportedProgram and launches
+    Model Explorer for interactive visualization. The visualization helps debug
+    and understand the graph structure after AutoDeploy transformations.
+    """
+
     def _apply(
         self,
         gm: GraphModule,
         cm: CachedSequenceInterface,
         factory: ModelFactory,
         shared_config: SharedConfig,
     ) -> Tuple[GraphModule, TransformInfo]:
-        PytorchExportedProgramAdapterImpl.print_tensor = print_tensor
-        PytorchExportedProgramAdapterImpl.add_outputs_metadata = add_outputs_metadata
+        """Export the graph and launch Model Explorer for visualization.
+
+        Args:
+            gm: The graph module to visualize.
+            cm: The cached sequence interface with input arguments.
+            factory: The model factory (unused).
+            shared_config: Shared configuration across transforms (unused).
+
+        Returns:
+            A tuple of the unchanged graph module and transform info indicating
+            whether visualization was successful or skipped.
+        """
+        if model_explorer is None:
+            return gm, TransformInfo(
+                skipped=True, num_matches=0, is_clean=True, has_valid_shapes=True
+            )
+
+        try:
+            # Export graph module to ExportedProgram for visualization
+            exported_program = te.export(gm, args=(), kwargs=cm.named_args, dynamic_shapes=None)
 
-        # TODO(yudong): make viz as non-block call.
-        ep = te.export(gm, args=cm.args, dynamic_shapes=cm.dynamic_shapes)
-        graph = ep.graph
-        # Ensure the ops land up in the right module for better viz
-        for n in graph.nodes:
-            if n.target in CUSTOM_OPS:
-                n.meta["nn_module_stack"] = n.args[0].meta["nn_module_stack"]
+            # Ensure custom ops appear nested under their parent modules
+            for node in exported_program.graph.nodes:
+                if node.target in CUSTOM_OPS:
```

Comment:
This list `CUSTOM_OPS` list is pretty outdated, we can probably deprecate it all together